### ISSUE TITLE
Incrementing strings

### ISSUE THREAD

**Issue by orklah** [vimeo/psalm#4195]
Hi!

I'm not too proud of this one but:
https://psalm.dev/r/85539c51a5

This can be really useful when working with Excel files to loop over column names. I'm fine with the "Possibly unintended string increment" issue, but `TypeDoesNotContainType` is not ideal here.

Thanks

**Comment by muglug** [comment]
Shuddering at that code, but a legit bug. This would actually be a good way for you (or someone else) to get to know Psalm's internals a bit – compare and contrasting how Psalm behaves when incrementing an `int` or `float`

**Comment by orklah** [comment]
Ok, I traced back the issue in the `IncDecExpressionAnalyzer::analyze` function. More precisely, I think the difference is in the condtion line 48 which will differentiate the string case from the rest.

In the rest, It will create a fake statement to simulate an addition with 1 to replace the PostInc.

My goal was to go in the string case and make the type loose it's details (for example, literal 'a' would become non-empty-string. I think I can even go further and transform any string into a non-empty-string.

But I was not able to understand what's to do in `NonDivArithmeticOpAnalyzer`. I saw the two imbricated foreach on line 176 that will iterate each possibility(and flag `$has_string_increment` when we iterate on a string at line 304. I guess that's where I have to change things, and I would have guessed that I have to change `$result_type` but the operation here seems good. The `$result_type` is not set so we just get the type string.
But then, later in the execution, Psalm seems to recover the literal 'a' out of nowhere and I'm kinda lost.

Any tips to help me understand how to override the literal 'a' into something less detailed? Or maybe I'm totally in the wrong direction!
